domain/application/state/application_test.go

```diff
@@ -237,6 +240,182 @@ func (s *applicationStateSuite) TestCreateApplicationWithCharmStoragePath(c *gc.
 	s.assertApplication(c, "666", platform, channel, scale, true)
 }
 
+// TestCreateApplicationWithResources tests creation of an application with
+// specified resources.
+// It verifies that the charm_resource table is populated, alongside the
+// resource and application_resource table with datas from charm and arguments.
+func (s *applicationStateSuite) TestCreateApplicationWithResources(c *gc.C) {
+	charmResources := map[string]charm.Resource{
+		"some-file": {
+			Name:        "foo-file",
+			Type:        "file",
+			Path:        "/some/path/foo.txt",
+			Description: "A file",
+		},
+		"some-image": {
+			Name: "my-image",
+			Type: "oci-image",
+			Path: "repo.org/my-image:tag",
+		},
+	}
+	rev := 42
+	addResourcesArgs := []application.AddApplicationResourceArg{
+		{
+			Name:   "foo-file",
+			Origin: charmresource.OriginUpload,
+		},
+		{
+			Name:     "my-image",
+			Revision: &rev,
+			Origin:   charmresource.OriginStore,
+		},
+	}
+
+	err := s.state.RunAtomic(context.Background(), func(ctx domain.AtomicContext) error {
```

Comment:
I know this is in tests, but should we still be using RunAtomic for new things?

Reply:
This is an update to a state method which already uses RunAtomic, not a new method.

Reply:
We're removing it if we touch it.

Reply:
To do that, it implies to update CreateApplication to take a context.Context instead of an AtomicContext. It feels like a specific PR, and may overlap with your works on application state.

We can also update the test to create a dumb AtomicContext which only embed a context, but it would break the layout of the test, making the entire removal of Atomic context in the file more painful (non regexable, because different pattern).

So I think it is out of scope.